### PR TITLE
9349 embed permissions

### LINKED ISSUE #9349: Review embed permissions on builder embed

### PR DESCRIPTION
**Pull request by javitonino**
Fixes #9349

### CODE REVIEW

**Comment by javitonino** [comment]
Please CR @juanignaciosl

File: app/controllers/carto/builder/public/embeds_controller.rb [hunk 1]
```diff
@@ -45,8 +45,11 @@ def load_vizjson
         end
 
         def ensure_viewable
-          return(render 'admin/visualizations/embed_map_error', status: 403) if @visualization.private?
-          return(render 'show_protected', status: 403) if @visualization.password_protected?
+          if @visualization.private? && !@visualization.has_read_permission?(current_viewer)
```
Comment (juanignaciosl) [inline app/controllers/carto/builder/public/embeds_controller.rb]: You can avoid this double check by reversing the order and using `is_viewable_by_user?`:

```ruby
if @visualization.password_protected?
  return(render 'show_protected', status: 403)
elsif !@visualization.is_viewable_by_user?(current_viewer)
   return(render 'admin/visualizations/embed_map_error', status: 403)
end
```

Not a big improvement though, up to you.

**Comment by juanignaciosl** [comment]
I've made a minor suggestion, up to you 👍

File: app/controllers/carto/builder/public/embeds_controller.rb (hunk 1 above)
Comment (javitonino) [inline app/controllers/carto/builder/public/embeds_controller.rb]: Aside from looking nicer, it is also more correct. By reading your comment I realised `private?` is not the same than `is_privacy_private?`. That should have been detected by the test suite, but there was one stub too much. Fixed, and thanks for the catch, that was a good one to spot.

**Pull request merged**